SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -89,37 +89,62 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
 		return;
 	}
 
-	UE_LOG(LogSpatialOSNetDriver, Log, TEXT("Loaded Map %s. Connecting to SpatialOS."), *LoadedWorld->GetName());
-
 	//checkf(!SpatialOSInstance->IsConnected(), TEXT("SpatialOS should not be connected already. This is probably because we attempted to travel to a different level, which current isn't supported."));
 
 	// Set the timer manager.
 	TimerManager = &LoadedWorld->GetTimerManager();
 
-	// Connect to SpatialOS.
-	Connect();
-
 	// Set up manager objects.
 	EntityRegistry = NewObject<UEntityRegistry>(this);
-}
 
-void USpatialNetDriver::Connect()
-{
+	// Handle connection configurations.
+	UE_LOG(LogSpatialOSNetDriver, Log, TEXT("Loaded Map %s. Connecting to SpatialOS."), *LoadedWorld->GetName());
+
 	if (Connection != nullptr)
 	{
 		return;
 	}
 
 	Connection = NewObject<USpatialWorkerConnection>();
+
+	if (LoadedWorld->URL.HasOption(TEXT("spatialLocator")))
+	{
+		Connection->LocatorConfig.ProjectName = LoadedWorld->URL.GetOption(TEXT("project="), TEXT(""));
+		Connection->LocatorConfig.DeploymentName = LoadedWorld->URL.GetOption(TEXT("deployment="), TEXT(""));
+		Connection->LocatorConfig.LoginToken = LoadedWorld->URL.GetOption(TEXT("token="), TEXT(""));
+		Connection->LocatorConfig.UseExternalIp = true;
+	}
+	else if (LoadedWorld->URL.HasOption(TEXT("spatialReceptionist")))
+	{
+		// Check for overrides in the travel URL.
+		if (!LoadedWorld->URL.Host.IsEmpty())
+		{
+			Connection->ReceptionistConfig.ReceptionistHost = LoadedWorld->URL.Host;
+			Connection->ReceptionistConfig.ReceptionistPort = LoadedWorld->URL.Port;
+		}
+		// Check if we need to use external IPs (if not connecting to localhost).
+		if (Connection->ReceptionistConfig.ReceptionistHost.Compare(TEXT("127.0.0.1")) == 0)
```

Comment:
Can this IP be moved to SpatialConstants.h?

Reply:
Sure, why not.